### ISSUE TITLE
Can't put host/service group in downtime when you aren't the contact for all hosts/services

### ISSUE THREAD

**Issue by tmcnag** [NagiosEnterprises/nagioscore#191]
> This issue has been automatically migrated from [http://tracker.nagios.org/view.php?id=0000387](http://tracker.nagios.org/view.php?id=0000387)  - Please follow that link for details.
> Originally submitted on 2012-10-10 by craiggenner

3.4.0 introduced the following (from the change log):

> - Users can now see hostgroups and servicegroups that contain at least one host or service they are authorized for, instead of having to be authorized for them all (Ethan Galstad)

The side affect of this is that if you want to put a host or service group into downtime you must be a contact for all the hosts or services in that group to be able to do it successfully.

For example, if you view the attached screen shot of a host group your will see that the MySQL group has 7 hosts in it, 4 of which I am a contact for (you can see the services count) and 3 which I am not. If I try and put the host group into downtime the following error message is returned:
"Sorry, but you are not authorized to commit the specified command."

While I can understand that you don't want to put hosts/services into downtime for which you are not a contact (and therefore not authorised), and I also understand the argument which means you can't tell people why.

So I'm wondering if it's possible to put hosts/services into downtime for which you are a contact, and then put up a message to that effect. I'm not sure how this would work, or what sort of message you would return to the user.

The same is also seen with:
- enable/disable notifications of all hosts in this host group
- enable/disable notifications of all services in this host/service group
- enable/disable active checks of all services in this host/service group

In other words you can't do anything to a host or service group through the interface if you aren't a contact for all the hosts or services in that group, you are denied.

**Comment by jfrickson** [comment]
At this point, it's impossible to perform a command on _some_ of the hosts/services in a host/service group. It _might_ be possible with nagios core 5.0.0, or it might not. For now, it would have to be done on a host-by-host or service-by-service basis, or have additional host/service groups that contain only items authorized for that user.

I'm changing the label from bug to enhancement and setting the milestone to 5.0.0.

**Comment by hedenface** [comment]
We can revisit this in the future if there is greater community feedback regarding the idea, until then I'm closing the issue.
